crates/apps_lib/src/cli.rs

```diff
@@ -4860,17 +4853,14 @@ pub mod args {
                 token,
                 amount,
             }];
-            let mut gas_spending_keys = vec![];
-            if let Some(key) = GAS_SPENDING_KEY.parse(matches) {
-                gas_spending_keys.push(key);
-            }
+            let gas_spending_key = GAS_SPENDING_KEY.parse(matches);
```

Comment:
Can we make `DISPOSABLE_SIGNING_KEY` require `GAS_SPENDING_KEY`? This is indeed required with the modifications of this PR but at the moment we don't have a constraint on the cli args. If people forget to manually pass the gas spending key they will stumble into a generic error while contracting the tx